src/mpi/init/mpir_init.c

```diff
@@ -221,6 +233,9 @@ int MPIR_Init_thread_impl(int *argc, char ***argv, int user_required, int *provi
         *provided = MPIR_ThreadInfo.thread_provided;
 
   fn_exit:
+    if (!p_session_ptr) {
+        MPL_atomic_store_int(&MPIR_world_model_state, MPICH_WORLD_MODEL_INITIALIZED);
+    }
     MPIR_INIT_UNLOCK;
```

Comment:
Any reason why we move state check of `MPIR_world_model_state` from line 204 to `fn_exit?`